code/src/App.js

```diff
@@ -1,9 +1,97 @@
-import React from 'react'
+import React, { useEffect, useState } from 'react'
+
+import ThoughtForm from './components/ThoughtForm'
+import ThoughtItem from './components/ThoughtItem'
+import Loading from './components/Loading'
+
+import { API_URL, API_LIKES } from './utils/urls'
+
+// Setting up state
 
 export const App = () => {
+  const [thoughts, setThoughts] = useState([]) 
+  const [newThought, setNewThought] = useState('')
+  const [loading, setLoading]  = useState(false)
+
+  // Mount app
+
+  useEffect(() => {
+    fetchThoughts()
+  }, [])
+
+  // Where loading-animation starts
+
+  const fetchThoughts = () => {
+    setLoading(true)
+    fetch(API_URL)
+    .then((res) => res.json())
+    .then((data) => setThoughts(data))
+    .finally(() => setLoading(false))
+
+    // Where loading ends (Page finished loading and thoughts are shown)
+  }
+
+
+  // Handles new thoughts 
+
+  const handleFormSubmit = (event) => {
+    event.preventDefault()
+
+    const options = {
+      method: 'POST',
+      headers: {
+        'Content-Type': 'application/json'
+      },
+      body: JSON.stringify({ message: newThought }),
+    }
+    event.target.reset()
+
+    fetch(API_URL, options)
+    .then((res) => res.json())
+    .then((data) => {
+    fetchThoughts()
+    })
+
+    // Clears input-field and let's you post a new thought
+    setNewThought('')
```

Comment:
Hm, I might be misunderstanding the code here, does this do the same as the function on 47? In that case, one of them can be removed.

Reply:
Haha, you might be right 😅 will check it out!